sbi/inference/posteriors/vi_posterior.py

```diff
@@ -1,69 +1,566 @@
 # This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
 # under the Affero General Public License v3, see <https://www.gnu.org/licenses/>.
-from functools import partial
-from math import ceil
-from typing import Any, Callable, Dict, List, Optional, Union
-from warnings import warn
+
+from copy import deepcopy
+from typing import Callable, Optional, Union, Iterable
 
 import numpy as np
 import torch
-import torch.distributions.transforms as torch_tf
-from pyro.infer.mcmc import HMC, NUTS
-from pyro.infer.mcmc.api import MCMC
-from torch import Tensor, nn
+from torch.distributions import Distribution
+
+from tqdm import tqdm
+
 
-from sbi import utils as utils
 from sbi.inference.posteriors.base_posterior import NeuralPosterior
-from sbi.samplers.rejection.rejection import rejection_sample
+from sbi.utils import mcmc_transform
 from sbi.types import Shape, TorchTransform
-from sbi.utils import del_entries
-from sbi.utils.torchutils import atleast_2d, ensure_theta_batched
+from sbi.utils.torchutils import (
+    atleast_2d_float32_tensor,
+    Tensor,
+    ensure_theta_batched,
+)
+
+from sbi.samplers.vi import (
+    adapt_variational_distribution,
+    check_variational_distribution,
+    make_object_deepcopy_compatible,
+    move_all_tensor_to_device,
+    get_flow_builder,
+    get_VI_method,
+    get_sampling_method,
+    get_quality_metric,
+)
 
 
 class VIPosterior(NeuralPosterior):
-    r"""Posterior $p(\theta|x)$ with `log_prob()` and `sample()` methods, obtained with
-    SNLE.<br/><br/>
-    SNLE trains a neural network to approximate the likelihood $p(x|\theta)$. The
-    `SNLE_Posterior` class wraps the trained network such that one can directly evaluate
-    the unnormalized posterior log probability $p(\theta|x) \propto p(x|\theta) \cdot
-    p(\theta)$ and draw samples from the posterior with MCMC.<br/><br/>
-    The neural network itself can be accessed via the `.net` attribute.
+    r"""Provides VI (Variational Inference) to sample from the posterior.<br/><br/>
+
```

Comment:
Could you put all formulas into latex style? I.e. add dollar signs and backslashes, `$p(\theta|x)$. It will display properly n our website then.